lib/view_component/base.rb

```diff
@@ -267,11 +269,11 @@ def view_cache_dependencies
       []
     end
 
-    # For caching, such as #cache_if
+    # Expected by ActionView::Template::Renderable
     #
     # @private
     def format
-      @__vc_variant if defined?(@__vc_variant)
+      nil
```

Comment:
I've spent some time investigating why this code returned the variant and I still don't really understand why. It was added during 1.x and it's been refactored numerous times. I don't think this has any meaningful contribution to caching anymore as Rails cache already uses the `LookupContext@details` ivar as an entry point to caching. Any cache generated from a ViewComponent render would already be keyed by the specific LookupContext details including locale, variants, formats, and handlers.